heudiconv/convert.py

```diff
@@ -849,6 +849,61 @@ def nipype_convert(
     return eg, prov_file
 
 
+def filter_partial_volumes(
+    nii_files: list[str],
+    bids_files: list[str],
+    bids_metas: list[dict[str, Any]],
+) -> Tuple[list[str] | str, list[str] | str, list[Any] | Any]:
+    """filter interrupted 4D scans volumes with missing slices on XA: see dcm2niix #742
+
+    Parameters
+    ----------
+    nii_files : list[str]
+        converted nifti filepaths
+    bids_files: list[str]
+        converted BIDS json filepaths
+    bids_metas : list[dict[str, Any]]
+        list of metadata dict loaded from BIDS json files
+
+    Returns
+    -------
+    nii_files
+        filtered niftis
+    bids_files
+        filtered BIDS jsons
+    bids_metas
+        filtered BIDS metadata
+
+    """
+    partial_volumes = [
+        not metadata.get("RawImage", True)
+        and metadata.get("SeriesNumber", 0) > 1000
```

Comment:
could you please add a comment above to point to current dcm2niix code which relies on this heuristic.

Meanwhile, I filed

- https://github.com/rordenlab/dcm2niix/issues/972